component/azstorage/azstorage.go

```diff
@@ -453,9 +453,17 @@ func (az *AzStorage) ReadInBuffer(options internal.ReadInBufferOptions) (length
 		return 0, nil
 	}
 
-	err = az.storage.ReadInBuffer(options.Handle.Path, options.Offset, dataLen, options.Data)
-	if err != nil {
-		log.Err("AzStorage::ReadInBuffer : Failed to read %s [%s]", options.Handle.Path, err.Error())
+	if options.Etag != nil {
+		etag, err := az.storage.ReadInBufferWithETag(options.Handle.Path, options.Offset, dataLen, options.Data)
+		if err != nil {
+			log.Err("AzStorage::ReadInBuffer : Failed to read %s [%s]", options.Handle.Path, err.Error())
+		}
+		*(options.Etag) = etag
```

Comment:
You can also use this,
`options.Etag = to.Ptr(etag)`

Reply:
No I do not want the pointer to etag, rather I have sent a pointer and I want to copy the value of etag to my variable.